pkg/container/init.go

```diff
@@ -0,0 +1,38 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Package container contains helpers for working with container engines
+// This package has no stability guarantees whatsoever!
+package container
+
+import (
+	"os/exec"
+)
+
+// Engine is the container engine to use to launch containers
+// Currently we support podman and docker.  Kind will use podman if it is
+// found in $PATH else it fall back to docker.
+var Engine string
+
+func init() {
+	for _, engine := range []string{"podman", "docker"} {
```

Comment:
i think this needs to be exposed from a CLI flag or via an environment variable (preferring the former).
having this precedence can be problematic if someone has podman installed but wants to use docker for kind instead.